src/shared/extensions.ts

```diff
@@ -7,3 +14,16 @@ export const VSCODE_EXTENSION_ID = {
     awstoolkit: 'amazonwebservices.aws-toolkit-vscode',
     python: 'ms-python.python'
 }
+
+/**
+ * Long-lived, extension-scoped, shared globals.
+ */
+export interface ExtContext extends vscode.ExtensionContext {
+    //extensionContext: vscode.ExtensionContext
+    awsContext: AwsContext
+    regionProvider: RegionProvider
+    settings: SettingsConfiguration
+    outputChannel: vscode.OutputChannel
+    telemetryService: TelemetryService
+    chanLogger: ChannelLogger
+}
```

Comment:
We already encapsulate some of this in ./src/shared/extensionGlobals.ts

Reply:
oh wow I missed that. `ext` has a totally different meaning in js, normally...